query/parse.go

```diff
@@ -320,14 +330,24 @@ func parseOperators(in []Q) (Q, error) {
 
 // parseExprList parses a list of query expressions. It is the
 // workhorse of the Parse function.
-func parseExprList(in []byte) ([]Q, int, error) {
+func parseExprList(in []byte, parenBalance *int) ([]Q, int, error) {
 	b := in[:]
 	var qs []Q
 	for len(b) > 0 {
 		for len(b) > 0 && isSpace(b[0]) {
 			b = b[1:]
 		}
+		// fmt.Println("PAREN BALANCE: ", parenBalance, string(b))
 		tok, _ := nextToken(b)
+		if tok != nil {
+			switch tok.Type {
+			case tokParenOpen:
+				(*parenBalance)++
+			case tokParenClose:
+				(*parenBalance)--
+			}
+		}
```

Comment:
lets do a tiny refactor here and move the below tok.Type checks into this tok != nil part. Just stick with if statements so the control flow around break vs continue below makes sense.